src/serac/coefficients/coefficient_extensions.hpp

```diff
@@ -25,30 +26,172 @@
  */
 namespace serac::mfem_ext {
 
+namespace detail {
+// methods for determining index type
+template <typename T>
+struct index_t {
+  using type = std::size_t;
+};
+
+template <typename T>
+struct index_t<mfem::Array<T>> {
+  using type = int;
+};
+
+// Methods for determining the size of a container
+template <typename T>
+auto size(T&& container)
+{
+  return container.size();
+}
+
+/// mfem:Array size call
+template <typename T>
+std::size_t size(mfem::Array<T> container)
+{
+  return container.Size();
+}
+
+// Methods for determining type of coefficient evaluations
+// Returns default POD-type
+template <typename T, typename = void>
+struct eval_result_t {
+  using type = T;
+};
+
+// returns return type for mfem::Coefficient
+template <typename T>
+struct eval_result_t<T, std::enable_if_t<std::is_base_of_v<mfem::Coefficient, T>>> {
+  using type = double;
+};
+
+// returns return type for mfem::VectorCefficient
+template <typename T>
+struct eval_result_t<T, std::enable_if_t<std::is_base_of_v<mfem::VectorCoefficient, T>>> {
+  using type = mfem::Vector;
+};
+
+/// methods for evaluating coefficient stuff with the same prototype
+double eval(double& d, mfem::ElementTransformation&, const mfem::IntegrationPoint&) { return d; }
+
+/// evaluates an mfem::Coefficient
+double eval(mfem::Coefficient& c, mfem::ElementTransformation& Tr, const mfem::IntegrationPoint& ip)
+{
+  return c.Eval(Tr, ip);
+}
+
+/// evaluates a mfem::VectorCoefficient
+mfem::Vector eval(mfem::VectorCoefficient& v, mfem::ElementTransformation& Tr, const mfem::IntegrationPoint& ip)
+{
+  mfem::Vector temp(v.GetVDim());
+  v.Eval(temp, Tr, ip);
+  return temp;
+}
+
+template <typename T, typename = void>
+struct is_iterable : std::false_type {
+};
+
+template <typename T>
+struct is_iterable<T, std::void_t<decltype(std::declval<T>().begin()), decltype(std::declval<T>().end())>>
+    : std::true_type {
+};
+
+template <typename T, typename = void>
+struct has_brackets : std::false_type {
+};
+
+template <typename T>
+struct has_brackets<T, std::void_t<decltype(std::declval<T>()[0])>> : std::true_type {
+};
+
+}  // namespace detail
+
 /**
- * @brief MakeTrueEssList takes in a FESpace, a vector coefficient, and produces a list
- *  of essential boundary conditions
+ * @brief This namespace provides a set of commonly used "digitizing" functions for convenience.
  *
- * @param[in] pfes A finite element space for the constrained grid function
- * @param[in] c A VectorCoefficient that is projected on to the mesh. All
- * d.o.f's are examined and those that are the condition (> 0.) are appended to
- * the vdof list.
- * @return The list of true dofs that should be part of the essential boundary conditions
+ * The goal of a "digizitizing" function is to convert a floating-point number to an integer to be used as an mfem
```

Comment:
```suggestion
 * The goal of a "digitizing" function is to convert a floating-point number to an integer to be used as an mfem
```

Though "digizitizing" sounds much more fun...